### PR TITLE
KAFKA-10106: log time taken to handle LeaderAndIsr request

### PR DESCRIPTION
**Pull request by rite2nikhil**
ReplicaManager!becomeLeaderOrFollower handles the LeaderAndIsr request, StateChangeLogger logs when this request is handled, however it can be useful to log when this calls ends and record the time taken, can help operationally. 

Proposal is to ReplicaManager!becomeLeaderOrFollower start measuring the time before the `replicaStateChangeLock` is acquired and log before the response is returned. 

Note lines that changed here are 1373-1377 and 1233, with the response being captured in 1246.

### CODE REVIEW

File: core/src/main/scala/kafka/server/ReplicaManager.scala [hunk 1]
```diff
@@ -1242,131 +1243,138 @@ class ReplicaManager(val config: KafkaConfig,
...
+              .setTopicName(tp.topic)
+              .setPartitionIndex(tp.partition)
+              .setErrorCode(error.code)
+          }.toBuffer
+          new LeaderAndIsrResponse(new LeaderAndIsrResponseData()
+            .setErrorCode(Errors.NONE.code)
+            .setPartitionErrors(responsePartitions.asJava))
+        }
       }
+      val endMs = time.milliseconds()
```
Comment (lbradstreet) [inline core/src/main/scala/kafka/server/ReplicaManager.scala:1373]: For ease of review, it's worth mentioning in the PR body that the only lines that changed here are 1373-1377 and 1233, with the response being captured in 1246.

**Review by lbradstreet — APPROVED** [review]
Non-binding LGTM. It will be helpful to have the timing recorded in the stateChangeLogger.

**Comment by ijuma** [comment]
ok to test

**Review by junrao — APPROVED** [review]
@rite2nikhil : Thanks for the PR. LGTM

**Pull request merged**